tests/test_hsm.py

```diff
@@ -366,6 +366,89 @@ def tearDown(self):
         del self.offset
         del self.xy0
 
+    @staticmethod
+    def computeDirectShapeFromGalSim(record, exposure, config):
+        """
+        Retrieve the shape as estimated directly by GalSim for comparison
+        purposes.
+
+        Parameters
+        ----------
+        record : `~lsst.afw.table.SourceRecord`
+            The record containing the center and footprint of the source which
+            needs measurement.
+        exposure : `~lsst.afw.image.Exposure`
+            The exposure containing the source which needs measurement.
+        config : `~lsst.meas.extensions.shapeHSM._hsm_shape.\
+                 HsmShape<Method>Config`
+            The configuration object containing parameters and settings for
+            this measurement. <Method> is the name of the correction method
+            being used (e.g., Ksb, Regauss, etc.).
+
+        Returns
+        -------
+        `~galsim.hsm.ShapeData`
+            An object containing the results of shape measurement.
+        """
+
+        # Get the center of the source as a Point2D.
+        center = geom.Point2D(record.get("centroid_x"), record.get("centroid_y"))
+
+        # Get the PSF image evaluated at the source centroid.
+        psfImage = exposure.getPsf().computeImage(center)
+        psfImage.setXY0(0, 0)
+
+        # Get the GalSim images to use in the EstimateShear call.
+        bbox = record.getFootprint().getBBox()
+        bounds = galsim.bounds.BoundsI(bbox.getMinX(), bbox.getMaxX(), bbox.getMinY(), bbox.getMaxY())
+        image = galsim.Image(exposure.image[bbox].array, bounds=bounds, copy=False)
+        psfBBox = psfImage.getBBox(afwImage.PARENT)
+        psfBounds = galsim.bounds.BoundsI(
+            psfBBox.getMinX(), psfBBox.getMaxX(), psfBBox.getMinY(), psfBBox.getMaxY()
+        )
+        psf = galsim.Image(psfImage.array, bounds=psfBounds, copy=False)
```

Comment:
To make this test a bit more non-trivial, remove `copy=False` everywhere. On the off chance if something that shouldn't change in-place changes, we could catch it.

Reply:
Done!